tests/test_modeling.py

```diff
@@ -72,3 +109,49 @@ def test_setfit_multilabel_classifier_chain_classifier_model_head():
     )
 
     assert type(model.model_head) is ClassifierChain
+
+
+def test_setfit_single_target_differentiable_head():
```

Comment:
I recently refactored the `test_trainer.py` tests into groups by using `unittest.TestCase`. This has the advantage of allowing us to download groups of models just once vs multiple times per test run. 

Would you be happy to group your tests in a similar way? 

(I realise the existing tests in `test_modeling.py` also need to be grouped - I'll do that later :))

Reply:
sure no problem 👍

Reply:
done!